adoc/chapters/programming_interface.adoc

```diff
@@ -12525,6 +12525,182 @@ be tested using the [code]#has_known_identity# trait class.
 include::{header_dir}/identity.h[lines=4..-1]
```

Comment:
I wanted to  make this comment on the previous paragraph that starts "Certain implementations of reductions can be made more efficient ...", but GitHub won't let me comment on an unchanged paragraph.

I wonder if the intent would be more clear if this paragraph started out like:

> Some of the overloads for the `reduction` function take an identity value and some do not.  An implementation is required to compute a correct reduction even when the application does not specify an identity value.  However, the implementation may be more efficient when the identity value is either provided by the application or is known by the implementation.  [...]

Reply:
I like this.  I've made the change in 954933ce3a9411fa7f21170803ff31d3da2b6622.  Since I was updating this paragraph, I also folded in @psalz's feedback from https://github.com/KhronosGroup/SYCL-Docs/issues/163#issuecomment-1240678038 to clarify that the identity depends on more than the binary operator alone.